biz.aQute.repository/src/aQute/bnd/repository/p2/provider/P2Indexer.java

```diff
@@ -93,14 +97,37 @@ File get(String bsn, Version version, Map<String,String> properties, DownloadLis
 		URI url = contentCapability.url();
 
 		final File source = client.getCacheFileFor(url);
+		final Path sourcePath = source.toPath();
 		final File link = new File(location, bsn + "-" + version + ".jar");
-		if (link.isFile())
-			Files.delete(link.toPath());
-		Files.createLink(link.toPath(), source.toPath());
+		final Path linkPath = link.toPath();
+
+		// links can fail if they span disparate Windows drives
+		// so we need to handle them differently
+		final AtomicBoolean createdSymlink = new AtomicBoolean();
+
+		if (link.isFile() && Files.isSymbolicLink(linkPath))
+			Files.delete(linkPath);
+
+		try {
+			Files.createLink(linkPath, sourcePath);
+			createdSymlink.set(true);
+		} catch (FileSystemException e) {
+			createdSymlink.set(false);
+		}
 
 		Promise<File> go = client.build().useCache(MAX_STALE).async(url.toURL()).map(new Function<File,File>() {
 			@Override
 			public File apply(File t) {
+				// if we can't create symlinks we need to copy this into link
```

Comment:
I guess I don't see why we cannot also defer the link into apply so that apply is create link or copy file. Then we don't need an AtomicBoolean.